### PR TITLE
Remove *.opam.template

### PR DESCRIPTION
**Pull request by smorimoto**

### CODE REVIEW

**Comment by hhugo** [comment]
The CI doesn't like this change. 
Could you try to motivate changes in the description whenever you open PRs ?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -23,62 +23,29 @@ jobs:
...
+            ocaml-compiler: "ocaml-variants.4.14.2+options,ocaml-option-32bit"
+          - os: ubuntu-latest
+            ocaml-compiler: "ocaml-variants.5.2.1+options,ocaml-option-32bit" # Note this OCaml compiler is bytecode only
 
     runs-on: ${{ matrix.os }}
 
     steps:
       - name: Set git to use LF
-        if: matrix.ocaml-compiler < 5.2
+        if: matrix.os == 'windows-latest'
```
Comment (hhugo) [inline .github/workflows/build.yml:48]: I don't understand this change. `ocaml-compiler < 5.2` was on purpose.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -109,41 +76,33 @@ jobs:
         uses: ocaml/setup-ocaml@v3
         with:
           ocaml-compiler: ${{ matrix.ocaml-compiler }}
-          dune-cache: true
-          opam-pin: false
-
-      - run: opam install conf-pkg-config
```
Comment (hhugo) [inline .github/workflows/build.yml:115]: We need this, this is a temporary fix for https://github.com/dra27/mingw-w64-shims/issues/1

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -109,41 +76,33 @@ jobs:
...
+        if: matrix.ocaml-compiler >= 5.0
 
       - run: opam exec -- git diff --exit-code
-        if: ${{ !matrix.skip-test }}
+        if: matrix.ocaml-compiler >= 4.14
 
-      - name: build doc
-        if: ${{ !matrix.skip-doc && github.event_name == 'push' && github.ref_name == 'master'}}
+      - name: Build doc
+        if: matrix.os == 'ubuntu-latest' && matrix.ocaml-compiler == '5.2' && github.event_name == 'push' && github.ref_name == 'master'
```
Comment (hhugo) [inline .github/workflows/build.yml:96]: I would prefer to only bump version once when it come to change the version we generate the doc with.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -23,62 +23,29 @@ jobs:
           - "4.11"
           - "4.12"
           - "4.13"
+          - "4.14"
           - "5.0"
           - "5.1"
-        skip-test:
```
Comment (hhugo) [inline .github/workflows/build.yml:28]: I find the previous version much easier to understand.

Comment (hhugo) [reply .github/workflows/build.yml:28]: This PR kind of inline the logic previously encoded by skip-* parameters. We know have to go over all step understand what's done for what version. I think this is going in the wrong direction.

**Comment by hhugo** [comment]
Can you please explain why you want to remove opam.template and try to keep the diff minimal in the CI

**Comment by smorimoto** [comment]
I was going to remove the excess as much as possible, but as you said, it was not as elegant as I expected.

**Pull request closed without merging**